src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/HashStream.cs

```diff
@@ -0,0 +1,172 @@
+using System.IO;
+using System.Security.Cryptography;
+using CoreWCF;
+using System;
+
+namespace CoreWCF.IdentityModel
+{
+    sealed class HashStream : Stream
+    {
+        HashAlgorithm hash;
+        long length;
+        bool disposed;
+        bool hashNeedsReset;
+        MemoryStream logStream;
+
+        /// <summary>
+        /// Constructor for HashStream. The HashAlgorithm instance is owned by the caller.
+        /// </summary>
+        public HashStream(HashAlgorithm hash)
+        {
+            if (hash == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("hash");
+
+            Reset(hash);
+        }
+
+        public override bool CanRead
+        {
+            get { return false; }
+        }
+
+        public override bool CanWrite
+        {
+            get { return true; }
+        }
+
+        public override bool CanSeek
+        {
+            get { return false; }
+        }
+
+        public HashAlgorithm Hash
+        {
+            get { return this.hash; }
+        }
+
+        public override long Length
+        {
+            get { return this.length; }
+        }
+
+        public override long Position
+        {
+            get { return this.length; }
+            set
+            {
+                // PreSharp Bug: Property get methods should not throw exceptions.
+#pragma warning suppress 56503
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new NotSupportedException());
+            }
+        }
+
+        public override void Flush()
+        {
+        }
+
+        public void FlushHash()
+        {
+            FlushHash(null);
+        }
+
+        public void FlushHash(MemoryStream preCanonicalBytes)
+        {
+
+            this.hash.TransformFinalBlock(CryptoHelper.EmptyBuffer, 0, 0);
+
+            //if (DigestTraceRecordHelper.ShouldTraceDigest)
```

Comment:
Can you add a comment here that this logs Pii data. I need to be able to search the "Pii" in the future to enable message logging and has the ability to log Pii data.